pkg/daemon/daemon.go

```diff
@@ -931,14 +941,14 @@ func (dn *Daemon) drainNode() error {
 		if err != nil {
 			lastErr = err
 			glog.Infof("Cordon failed with: %v, retrying", err)
-			return false, nil
+			return false, err
 		}
 		err = drain.RunNodeDrain(dn.drainer, dn.name)
 		if err == nil {
-			return true, nil
+			glog.Infof("Draining failed with: %v, retrying", err)
```

Comment:
if err is nil it means drain succeeded so original code was good imo

Reply:
it makes sense, I'll revert this change